### DISCUSSION TITLE
Fatal Error: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory

### DISCUSSION THREAD

**Discussion by gchacaltana** [RocketChat/Rocket.Chat#31323] [Q&A]
Hi team, I'm using Rocket Chat (version 5.4.0) on a Linux instance, running on a workload with 4 GB of RAM, node version 14.19, and we have identified that when we have memory consumption peaks greater than that reach 2 GB, an error with the following message is sometimes generated: "FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory", however, the workload is configured to 4 GB of RAM. It is as if the Rocket Chat instance cannot consume all the memory. Has anything similar happened to you? Do you think it is a Rocket Chat configuration or the way the application is run with Node?

**Comment by florisvangeel** [comment]
Just noticed this issue trying to build `tags/7.0.0` from git on ubuntu 24.04 (no docker)
These heaps are used to bundle and compress javascript, most likely this happened in your build process.

Besides that 7.x now requires [deno](https://docs.deno.com/runtime/getting_started/installation/) 
The javascript compressed file sizes also increased.
my build LXC has 16gb, so in this part of this process i set those to half the memory:
```
export NODE_OPTIONS="--max-old-space-size=8192"
```
Not sure how much is minimally needed, but it ran with 8gb.
After the `meteor build` runs fine.

**Comment by florisvangeel** [comment]
Update: in order to build 7.6.x and above 16GB of memory isn't enough anymore.
So i upped the build machine to 20GB internal memory.
